lib/asciidoctor/extensions.rb

```diff
@@ -440,6 +440,71 @@ def default_attrs value
       option :default_attrs, value
     end
     alias :seed_attributes_with :default_attrs
+
+    def resolves_attributes *args
+      args = args[0] unless args.size > 1
+      case args
+      when false
+        option :content_model, :text
+        return
+      when nil, true
+        option :pos_attrs, []
+        option :default_attrs, {}
+      when ::Array
+        names, defaults = [], {}
+        args.each do |arg|
+          if arg.include? '='
+            name, value = arg.split '=', 2
+            if name.include? ':'
+              idx, name = name.split ':', 2
+              idx = idx == '@' ? names.size : idx.to_i
+              names[idx] = name
+            end
+            defaults[name] = value
+          elsif arg.include? ':'
+            idx, name = arg.split ':', 2
+            idx = idx == '@' ? names.size : idx.to_i
+            names[idx] = name
+          else
+            names << arg
+          end
+        end
+        option :pos_attrs, names.compact
+        option :default_attrs, defaults
+      when ::String
+        if args.include? '='
+          name, value = args.split '=', 2
+          if name.include? ':'
+            _, args = args.split ':', 2
+            option :pos_attrs, [args]
+          else
+            option :pos_attrs, []
+          end
+          option :default_attrs, { name => value }
+        else
+          _, args = args.split ':', 2 if args.include? ':' 
```

Comment:
You are correct. I'm in the process of fixing this bug. The code got messed up when I moved some things around.